consensus/src/block_storage/block_store.rs

```diff
@@ -476,18 +518,27 @@ impl BlockReader for BlockStore {
         self.inner.read().highest_quorum_cert()
     }
 
-    fn highest_commit_cert(&self) -> Arc<QuorumCert> {
-        self.inner.read().highest_commit_cert()
+    fn highest_ordered_cert(&self) -> Arc<QuorumCert> {
+        self.inner.read().highest_ordered_cert()
+    }
+
+    fn highest_ledger_info(&self) -> LedgerInfoWithSignatures {
+        self.inner.read().highest_ledger_info()
     }
 
     fn highest_timeout_cert(&self) -> Option<Arc<TimeoutCertificate>> {
         self.inner.read().highest_timeout_cert()
     }
 
     fn sync_info(&self) -> SyncInfo {
-        SyncInfo::new(
+        SyncInfo::new_decoupled(
             self.highest_quorum_cert().as_ref().clone(),
-            self.highest_commit_cert().as_ref().clone(),
+            self.highest_ordered_cert().as_ref().clone(),
+            if self.highest_ledger_info().commit_info().round() > 0 {
```

Comment:
this filter is already done inside constructor?

Reply:
resolved